### PR TITLE
feat(recent-search): List recent searches in Issues search bar [SEN-352]

### PR DESCRIPTION
**Pull request by billyvg**
It's at the top here because the items below are not interact-able. Will need some design/icon work from @Chrissy as a follow up. Also API does not currently filter by current search input.

image

## Tag Key
image

## Tag Value
image

Fixes SEN-352

### CODE REVIEW

File: src/sentry/static/sentry/app/components/smartSearchBar.jsx [hunk 1]
```diff
@@ -437,14 +510,26 @@ class SmartSearchBar extends React.Component {
 
       searchItems[state.activeSearchItem].active = true;
       this.setState({searchItems: searchItems.slice(0)});
-    } else if (evt.key === 'Tab' && !this.isDefaultDropdown()) {
+    } else if (evt.key === 'Tab') {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/smartSearchBar.jsx]: So we can navigate to recent items using keyboard.

File: src/sentry/static/sentry/app/components/smartSearchBar.jsx [hunk 2]
```diff
@@ -437,14 +510,26 @@ class SmartSearchBar extends React.Component {
...
   };
 
-  onAutoComplete = replaceText => {
+  onAutoComplete = (replaceText, item) => {
+    if (item.type === 'recent-search') {
+      this.setState({query: replaceText}, () => {
+        // Propagate onSearch and save to recent searches
+        this.doSearch();
+      });
+
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/smartSearchBar.jsx]: Will follow up with analytics here.

File: src/sentry/static/sentry/app/components/smartSearchBar.jsx [hunk 3]
```diff
@@ -356,28 +414,29 @@ class SmartSearchBar extends React.Component {
...
-      return (tag.predefined ? this.getPredefinedTagValues : this.getTagValues)(
-        tag,
-        preparedQuery,
-        this.updateAutoCompleteState
-      );
+      const fetchTagValuesFn = tag.predefined
+        ? this.getPredefinedTagValues
+        : this.getTagValues;
+      const tagValues = await fetchTagValuesFn(tag, preparedQuery);
+      const recentSearches = await this.getRecentSearches();
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/smartSearchBar.jsx]: Should these be run in serial? Seems like a place to use Promise.all()

File: src/sentry/static/sentry/app/components/smartSearchBar.jsx [hunk 4]
```diff
@@ -153,18 +163,29 @@ class SmartSearchBar extends React.Component {
...
-      // Ignore errors if it fails to save
+      try {
+        await saveRecentSearch(api, orgId, recentSearchType, query);
+
+        if (onSavedRecentSearch) {
+          onSavedRecentSearch(query);
+        }
+      } catch (err) {
+        // Ignore errors if it fails to save
+        Sentry.captureException(err);
```
[the remark this answers is not in the document]
Comment (billyvg) [reply src/sentry/static/sentry/app/components/smartSearchBar.jsx]: Updated comment to be more clear

**Review by markstory — APPROVED** [review]

**Pull request merged**